RIOT-OS/RIOT
# cpu/atmega_common: Split into atmega_common and avr8_common

### Contribution description

The current RIOT-OS have been supporting AVR mega only MCUs. This intents to pave the road to allow add other variants like ATtiny or even ATxmega. The way to do it is moving AVR-8 CPU common code from ATmega specific code.

The **clock**, **BADISR_vect** and **reset cause** should be reworked first by creating two new files. This makes a way to move all AVR-8 common code from **atmega_common** to **avr8_common** with, almost, a simple rename. The last part is rework the **atmega_** prefix to **avr8_** prefix.

### Testing procedure

Run `make BOARD=<board> -C examples/hello-world all` to ensure that no build issues were created.

### Issues/PRs references

The XMEGA feature request with initial discussion about this was opened at #15703. It shares two branches: 1) this PR itself; 2) a XMEGA prove of concept that runs into two XMEGA CPU variants A1/A3. The execution test from [2] can be made by running `Tutorials/task-01` project, for instance.

CC: @maribu
